MessengerOnFileForLinux/Common/src/SignalHandling.cpp

```diff
@@ -33,36 +31,55 @@ std::string getChatFolderName()
     return "";
 }
 
-void sigintHandlerInMainConsole(int signal)
+void closeMessegner()
 {
-    fileLog("SIGINT handled in main console", LogSpace::Common);
-    signalStatus = signal;
     SignOut signOut;
     signOut.signOutUser();
     exit (EXIT_SUCCESS);
 }
+}//namespace
+
+void posixSignalHandlerInMainConsole(int signal)
+{
+    const std::string log = "Signal nr=" + std::to_string(signal) + "handled in main console";
+    fileLog(log.c_str(), LogSpace::Common);
+    if (not isChatResourcesDealocated)
+    {
+        isChatResourcesDealocated = true;
+        fileLog("Start of remove chat resources", LogSpace::Common);
+        closeMessegner();
+    }
+}
 
-void sigintHandlerInChatConsole(int signal)
+void posixSignalHandlerInChatConsole(int signal)
 {
-    fileLog("SIGINT handled in chat console", LogSpace::Common);
-    signalStatus = signal;
-    std::string chatFolderName = getChatFolderName();
-    fileLog(chatFolderName.c_str(), LogSpace::Common);
+    const std::string log = "Signal nr=" + std::to_string(signal) + "handled in chat console";
+    fileLog(log.c_str(), LogSpace::Common);
 
-    if (FileInterface::Managment::isFileExist(ENVIRONMENT_PATH::TO_FOLDER::CHATS + chatFolderName + "/END"))
+    if (not isChatResourcesDealocated)
     {
-        std::string command = "rm -rf " + ENVIRONMENT_PATH::TO_FOLDER::CHATS + chatFolderName;
-        system(command.c_str());
-        SignOut signOut;
-        signOut.signOutUser();
-        exit (EXIT_SUCCESS);
+        isChatResourcesDealocated = true;
+        fileLog("Start of remove chat resources", LogSpace::Common);
+        std::string chatFolderName = getChatFolderName();
+        if (FileInterface::Managment::isFileExist(ENVIRONMENT_PATH::TO_FOLDER::CHATS + chatFolderName + "/END"))
+        {
+            std::string command = "rm -rf " + ENVIRONMENT_PATH::TO_FOLDER::CHATS + chatFolderName;
+            system(command.c_str());
+            closeMessegner();
```

Comment:
duplikacja kodu z elsem, wywalić poza ifa

Reply:
Tak, ale to jest jedna linijka i moim zdaniem duzo lepiej tak czyta sie ten kod. Jak spróbowałem zmienić, to nie wygląda to dobrze

Reply:
nie zgadzam się, czemu niby lepiej kaurat wygląda jedna linijka w dwóch if elsach zamiast na końcu bez powtórzenia?